infrastructures/infrastructure-azure/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/AzureInfrastructure.java

```diff
@@ -193,130 +169,131 @@
+        this.domain = getParameter(parameters, Indexes.DOMAIN.index);
+        this.subscriptionId = getParameter(parameters, Indexes.SUBSCRIPTION_ID.index);
+        this.authenticationEndpoint = getParameter(parameters, Indexes.AUTHENTICATION_ENDPOINT.index);
+        this.managementEndpoint = getParameter(parameters, Indexes.MANAGEMENT_ENDPOINT.index);
+        this.resourceManagerEndpoint = getParameter(parameters, Indexes.RESOURCE_MANAGER_ENDPOINT.index);
+        this.graphEndpoint = getParameter(parameters, Indexes.GRAPH_ENDPOINT.index);
+        this.rmHostname = getParameter(parameters, Indexes.RM_HOSTNAME.index);
+        this.connectorIaasURL = getParameter(parameters, Indexes.CONNECTOR_IAAS_URL.index);
+        this.image = getParameter(parameters, Indexes.IMAGE.index);
+        this.imageOSType = getParameter(parameters, Indexes.IMAGE_OS_TYPE.index).toLowerCase();
+        this.vmSizeType = getParameter(parameters, Indexes.VM_SIZE_TYPE.index);
+        this.vmUsername = getParameter(parameters, Indexes.VM_USERNAME.index);
+        this.vmPassword = getParameter(parameters, Indexes.VM_PASSWORD.index);
+        this.vmPublicKey = getParameter(parameters, Indexes.VM_PUBLIC_KEY.index);
+        this.resourceGroup = getParameter(parameters, Indexes.RESOURCE_GROUP.index);
+        this.region = getParameter(parameters, Indexes.REGION.index);
+        this.numberOfInstances = Integer.parseInt(getParameter(parameters, Indexes.NUMBER_OF_INSTANCES.index));
+        this.numberOfNodesPerInstance = Integer.parseInt(getParameter(parameters,
+                                                                      Indexes.NUMBER_OF_NODES_PER_INSTANCE.index));
+        this.nodeJarURL = getParameter(parameters, Indexes.NODE_JAR_URL.index);
+        this.privateNetworkCIDR = getParameter(parameters, Indexes.PRIVATE_NETWORK_CIDR.index);
+        this.staticPublicIP = Boolean.parseBoolean(getParameter(parameters, Indexes.STATIC_PUBLIC_IP.index));
+        this.additionalProperties = getParameter(parameters, Indexes.ADDITIONAL_PROPERTIES.index);
+        this.resourceUsageRefreshFreqInMin = Integer.parseInt(getParameter(parameters,
+                                                                           Indexes.BILLING_RESOURCE_USAGE_REFRESH_FREQ_IN_MIN.index));
+        this.rateCardRefreshFreqInMin = Integer.parseInt(getParameter(parameters,
+                                                                      Indexes.BILLING_RATE_CARD_REFRESH_FREQ_IN_MIN.index));
+        this.offerId = getParameter(parameters, Indexes.BILLING_OFFER_ID.index);
+        this.currency = getParameter(parameters, Indexes.BILLING_CURRENCY.index);
+        this.locale = getParameter(parameters, Indexes.BILLING_LOCALE.index);
+        this.regionInfo = getParameter(parameters, Indexes.BILLING_REGION_INFO.index);
+        this.maxBudget = Double.parseDouble(getParameter(parameters, Indexes.BILLING_BUDGET.index));
```

Comment:
no need to catch exceptions with ur `parseInt` and `parseDouble` operations?

Reply:
You re right but I strictly followed the existing behavior.. Exemple in the same function  "this.numberOfInstances = Integer.parseInt(getParameter(parameters, Indexes.NUMBER_OF_INSTANCES.index));"

Reply:
Ok

Reply:
Ok